src/test/java/mate/academy/security/CustomUserDetailsServiceTest.java

```diff
@@ -0,0 +1,57 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+class CustomUserDetailsServiceTest {
+    private static final String EMAIL = "bchupika@mate.academy";
+    private UserDetailsService userDetailsService;
+    private UserService userService;
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        userService = Mockito.mock(UserService.class);
+        userDetailsService = new CustomUserDetailsService(userService);
+        user = new User();
+        user.setEmail(EMAIL);
+        user.setPassword("12345678");
+        user.setRoles(Set.of(new Role(Role.RoleName.USER)));
+
+    }
+
+    @Test
+    void loadUserByUsername_Ok() {
+
+        Mockito.when(userService.findByEmail(EMAIL)).thenReturn(Optional.of(user));
+
+        UserDetails actual = userDetailsService.loadUserByUsername(EMAIL);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(EMAIL, actual.getUsername());
+        Assertions.assertEquals("12345678", actual.getPassword());
+    }
+
+    @Test
+    void loadUserByUsername_UsernameNotFound() {
+        Mockito.when(userService.findByEmail(EMAIL)).thenReturn(Optional.of(user));
+
+        try {
+            userDetailsService.loadUserByUsername("user@user");
+        } catch (UsernameNotFoundException e) {
+            Assertions.assertEquals("User not found.", e.getMessage());
+            return;
+        }
+
+        Assertions.fail("Expected to receive UserNotFoundException");
```

Comment:
We almost never use try catch in tests. Here, you can use this approach
```suggestion
    Exception exception = assertThrows(UsernameNotFoundException.class, () -> {
        userDetailsService.loadUserByUsername("user@user");
    });

    assertEquals("User not found.", exception.getMessage());
```